src/lib.rs

```diff
@@ -195,8 +195,8 @@ mod iter;
 mod shard;
 use cfg::CfgPrivate;
 pub use cfg::{Config, DefaultConfig};
-
 pub use pool::Pool;
```

Comment:
I think we need to reexport the `PoolGuard` type as well, since the `pool` module is now private:
```suggestion
pub use pool::{Pool, PoolGuard};
```
otherwise, users won't be able to reference it.